server-adapters/resteasy-reactor-netty/src/main/java/org/jboss/resteasy/plugins/server/reactor/netty/ReactorNettyJaxrsServer.java

```diff
@@ -0,0 +1,319 @@
+       start();
+       server.onDispose().block();
+   }
+
+   class Handler {
+
+      private final Mono<InputStream> empty = Mono.just(new InputStream() {
+         @Override
+         public int read() {
+            return -1;  // end of stream
+         }
+      });
+
+      Publisher<Void> handle(final HttpServerRequest req, final HttpServerResponse resp) {
+
+         if (1 == 2) {
+            final NettyOutbound o =
+                resp.send(req.receive().retain().doOnEach(b -> log.trace("got some bytes!")));
+            return o.then()
+                .doFinally(s -> log.trace("Request processing finished with: {}", s));
+         }
+
+         final ResteasyUriInfo info = new ResteasyUriInfo(req.uri(), root);
+
+         // aggregate (and maybe? asInputStream) reads the entire request body into memory (direct?)
+         // Can we stream it in some way?
+         // https://stackoverflow.com/a/51801335/2071683 but requires a thread.  Isn't using a thread
+         // per request even if from the elastic pool a big problem???  I mean we are trying to reduce
+         // threads!
+         // I honestly don't know what the Netty4 adapter is doing here.  When
+         // I try to send a large body it says "request payload too large".  I
+         // don't know if that's configurable or not..
+        
+
+         // This is a subscription tied to the completion writing the response.
+         final MonoProcessor<Void> completionMono = MonoProcessor.create();
+
+         return req.receive()
+             .aggregate()
+             .asInputStream()
+             .switchIfEmpty(empty)
+             .flatMap(body -> {
+                log.trace("Body read!");
+                resp.withConnection(conn -> {
+                   log.trace("{}", conn.channel().id());
+                });
+
+                // These next 2 classes provide the main '1-way bridges' between reactor-netty and RestEasy.
+                final ReactorNettyHttpResponse resteasyResp =
+                    new ReactorNettyHttpResponse(req.method(), resp, completionMono);
+
+                final ReactorNettyHttpRequest resteasyReq =
+                    new ReactorNettyHttpRequest(
+                        info,
+                        req,
+                        body,
+                        resteasyResp,
+                        (SynchronousDispatcher) deployment.getDispatcher()
+                    );
+
+                // This is what actually kick RestEasy into action.
+                ResteasyProviderFactory defaultInstance = ResteasyProviderFactory.getInstance();
+                if (defaultInstance instanceof ThreadLocalResteasyProviderFactory) {
+                   ThreadLocalResteasyProviderFactory.push(deployment.getProviderFactory());
+                }
+
+                deployment.getDispatcher().invoke(resteasyReq, resteasyResp);
+
+                if (defaultInstance instanceof ThreadLocalResteasyProviderFactory) {
```

Comment:
Somewhere before this line, we should be setting the `sslContext` it seems. I think follow up PRs will address it